lib/Sema/SemaResolveInherit.cpp

```diff
@@ -0,0 +1,400 @@
+    auto pDeclIt = std::find_if(BaseDecl.begin(), BaseDecl.end(),
+                                [cDecl](const Decl *node) {
+                                  return node->getName() == cDecl->getName();
+                                });
+
+    if (pDeclIt == BaseDecl.end()) {
+      if (AppendChild)
+        BaseDecl.emplace_back(cDecl);
+      continue;
+    }
+
+    if (auto pDecl = dynamic_cast<FunctionDecl *>(*pDeclIt)) {
+      // check cDecl is marked override
+      // TODO: function can be overrided by status var....
+      if (auto cFDecl = dynamic_cast<FunctionDecl *>(cDecl)) {
+        if (cFDecl->getOverrideSpecifier() == nullptr) {
+          Diag(cFDecl->getLocation().getBegin(),
+               diag::err_decl_need_to_be_overrided)
+              << pDecl->getName();
+        } else {
+          // TODO: check compatiblity for both cont
+        }
+      } else {
+        Diag(cFDecl->getLocation().getBegin(),
+             diag::err_func_to_var_overrided_unsoupport);
+      }
+
+      // check pDecl is virtual
+      if (!pDecl->isVirtual()) {
+        Diag(pDecl->getLocation().getBegin(), diag::err_decl_need_to_be_virtual)
+            << pDecl->getName();
+      }
+
+      // check visibility
+      // only external to public is allowed
+      if (!(pDecl->getVisibility() == cDecl->getVisibility() ||
+            (pDecl->getVisibility() == Decl::Visibility::External &&
+             cDecl->getVisibility() == Decl::Visibility::Public))) {
+        Diag(cDecl->getLocation().getBegin(), diag::err_visibility_not_match)
+            << pDecl->getName();
+      }
+
+      // check mutability
+      {
+        bool allowed = false;
+        auto cFDecl = dynamic_cast<FunctionDecl *>(cDecl);
+        assert(cFDecl);
+        switch (pDecl->getStateMutability()) {
+        case StateMutability::NonPayable:
+          allowed |=
+              cFDecl->getStateMutability() == StateMutability::NonPayable;
+          [[fallthrough]];
+        case StateMutability::View:
+          allowed |= cFDecl->getStateMutability() == StateMutability::View;
+          [[fallthrough]];
+        case StateMutability::Pure:
+          allowed |= cFDecl->getStateMutability() == StateMutability::Pure;
+          break;
+        case StateMutability::Payable:
+          allowed |= cFDecl->getStateMutability() == StateMutability::Payable;
+          break;
+        default:
+          __builtin_unreachable();
+        }
+        if (!allowed) {
+          Diag(cDecl->getLocation().getBegin(),
+               diag::err_mutability_overriding_not_allow)
+              << cDecl->getName();
+        }
+      }
+
+      BaseDecl.erase(pDeclIt);
+    } else if (auto pDecl = dynamic_cast<VarDecl *>(*pDeclIt)) {
+      Diag(cDecl->getLocation().getBegin(),
+           diag::err_statevar_cannot_be_overrided)
+          << cDecl->getName();
+    }
+
+    if (AppendChild)
+      BaseDecl.emplace_back(cDecl);
+  }
+
+  // todo: the order of the Decl is important?
```

Comment:
The order of Decl is important I think.
The state variables will use the order in hashing their storage address in the ethereum storage, AFAIK.